### ISSUE TITLE
Dataset

### ISSUE THREAD

**Issue by cs123951** [vincentme/GroupRegNet#1]
Hi,

Thank you for sharing such brilliant work. I have read your paper _GroupRegNet: A Groupwise One-shot Deep Learning-based 4D Image Registration Method_. The results on DIRLab and POPI is better than other deep-learning-based methods.  But the paper did not mention the training dataset and the validation dataset. I am curious about what is the training dataset you used. 
     Hope you could answer my question.
     Thank you very much.

**Comment by vincentme** [comment]
Hi 

This GroupRegNet is a totally unsupervised one-shot registration method. Therefore, the images to be registered are the only data fed into the neural network. So it makes no sense to differentiate the training/eval/testing dataset here. Hope this makes it clear.

**Comment by cs123951** [comment]
Thank you for your quick reply.
Machine learning algorithms are different than iterative algorithms. The parameters of machine learning are fixed once the training process ends.
If the images to be registered are the only data fed into the neural network, then how do we get the parameters of the neural network? Do we need to update the parameters iteratively?
If the parameters are determined by the registered lung data, then these data can not be used to evaluate the performance of the model. For example, If we give a child some dog's pictures to tell him what is a dog, how do we test if the child has learned what the dog is? We will show him other pictures that he has never seen. 
In the book of _Deep Learning[1]_, it is said that,
Page 110, "The central challenge in machine learning is that we must perform well on new, previously unseen inputs—not just those on which our model was trained. The ability to perform well on previously unobserved inputs is called generalization ."
"We typically estimate the generalization error of a machine learning model by measuring its performance on a test set of examples that were collected separately from the training set."

[1] Goodfellow, Ian, Yoshua Bengio, and Aaron Courville. 2016. Deep Learning.

**Comment by vincentme** [comment]
Yes, the procedure of a typical training and inference learning algorithm behaves as you described. However, this particular registration algorithm does not follow that framework. As discussed in the paper 

> From a broader perspective, GroupRegNet can be viewed as a mixture of conventional and learning-based methods. It follows the same iterative optimization process of the conventional approach and only uses the images to be registered as input. Furthermore, segmentation images, annotated landmarks, or deformation fields do not need to be provided to the neural network. GroupRegNet utilizes CNN as the motion model whose weights are learned through optimization. The performance improvement over the conventional approach can be attributed to the more expressive power of the deep neural network and to fewer assumptions in the DVF. Comparing to a typical training and inference procedure of learning-based methods, the one-shot learning strategy eliminates the requirement of abundant training images and annotations, thus improves the accuracy. The problem of over-fitting for the one-shot training strategy was not presented due to the well regularized total loss.

**Comment by cs123951** [comment]
Thank you for your kind reply, I have figured it out.
